flake.nix

```
@@ -50,18 +47,24 @@
           pname = "niri";
           version = self.shortRev or self.dirtyShortRev or "unknown";
 
-          src = nix-filter.lib.filter {
-            root = self;
-            include = [
-              "niri-config"
-              "niri-ipc"
-              "niri-visual-tests"
-              "resources"
-              "src"
-              ./Cargo.lock
-              ./Cargo.toml
-            ];
-          };
+          src =
+            let
+              fs = lib.fileset;
+            in
+            fs.toSource {
+              root = ./.;
+              fileset = fs.intersection (fs.fromSource (lib.sources.cleanSource ./.)) (
+                fs.unions [
+                  ./niri-config
+                  ./niri-ipc
+                  ./niri-visual-tests
+                  ./resources
+                  ./src
+                  ./Cargo.toml
+                  ./Cargo.lock
+                ]
+              );
+            };
```

what's with the `lib.sources.cleanSource`? this removes VCS noise; but we're in a git flake; `./.` is already "cleaned" and we actually *can't* even access the files that normally would have been cleaned by that function. it does *nothing*.


```suggestion
          src = lib.fileset.toSource {
            root = ./.;
            fileset = lib.fileset.unions [
              ./niri-config
              ./niri-ipc
              ./niri-visual-tests
              ./resources
              ./src
              ./Cargo.toml
              ./Cargo.lock
            ];
          };
```

both the original `nix-filter`, the version intersecting `cleanSource`, and this one with just `unions`, produce the *exact same store path*.